core/utils/dmlresult.py

```diff
@@ -0,0 +1,71 @@
+from core.utils.dmljob import serialize_job, deserialize_job
+
+class DMLResult(object):
+    """
+    DML Result
+
+    This class is created by the Runners whenever they're done running a job,
+    and it's passed to the Communication Manager on success or back to the
+    Scheduler on error.
+
+    TODO: Right now the DMLResult object is passing around the DMLJob that
+    originated the results, but we can do better.
+
+    """
+
+    def __init__(
+        self,
+        status,
+        job_type,
+        results={},
+        error_message="",
+    ):
+        """
+        Initializes a DML Result object.
+
+        Args:
+            status (str): status of the job [successful|failed].
```

Comment:
Nitpick: if status is only going to be `successful` or `failed`, why not just use a boolean?

Reply:
more statuses in future